python/cudf/cudf/core/column/column.py

```diff
@@ -1852,7 +1888,10 @@ def as_column(arbitrary, nan_as_null=None, dtype=None, length=None):
                             )
                         return as_column(data, nan_as_null=nan_as_null)
                     dtype = pd.api.types.pandas_dtype(dtype)
-                    if is_categorical_dtype(dtype):
+                    if (
+                        is_categorical_dtype(dtype)
+                        or type(dtype) == pd.core.dtypes.dtypes.IntervalDtype
```

Comment:
Can we use `is_interval_dtype` here?

Reply:
Yes! I hadn't realised I could add this for pandas types too, I'll do that now :D

Reply:
I believe it will work for Pandas interval types too, since they have `dtype.name == "interval"` (as we do)

Reply:
Yes, you're right, I've made adjustments to the code and tests seem to be passing :D